sktime/transformations/base.py

```diff
@@ -48,93 +104,567 @@ class BaseTransformer(BaseEstimator):
+
+        # treating Series vs Panel conversion for X
+        ###########################################
+
+        # there are three cases to treat:
+        # 1. if the internal _fit supports X's scitype, move on to mtype conversion
+        # 2. internal only has Panel but X is Series: consider X as one-instance Panel
+        # 3. internal only has Series but X is Panel:  loop over instances
+        #     currently this is enabled by conversion to df-list mtype
+        #     and this does not support y (unclear what should happen here)
+
+        # 1. nothing to do - simply don't enter any of the ifs below
+        #   the "ifs" for case 2 and 3 below are skipped under the condition
+        #       X_input_scitype in X_inner_scitypes
+        #   case 2 has an "else" which remembers that it wasn't entered
+
+        # 2. internal only has Panel but X is Series: consider X as one-instance Panel
+        if (
+            X_input_scitype == "Series"
+            and "Series" not in X_inner_scitypes
+            and "Panel" in X_inner_scitypes
+        ):
+            # convert the Series X to a one-element Panel
+            X = convert_Series_to_Panel(X)
+            # remember that we converted the Series to a one-element Panel
+            X_was_Series = True
+        else:
+            # remember that we didn't convert a Series to a one-element Panel
+            X_was_Series = False
+
+        # 3. internal only has Series but X is Panel: loop over instances
+        if (
+            X_input_scitype == "Panel"
+            and "Panel" not in X_inner_scitypes
+            and "Series" in X_inner_scitypes
+        ):
+            if y is not None:
+                ValueError(
+                    "no default behaviour if _fit does not support Panel, "
+                    " but X is Panel and y is not None"
+                )
+            X = convert_to(X, to_type="df-list", as_scitype="Panel")
+
+            if self.get_tag("fit-in-transform"):
+                Xt = [clone(self).transform(Xi) for Xi in X]
+            else:
+                transformers = self.transformers_
+                if len(transformers) != len(X):
+                    raise RuntimeError(
+                        "found different number of instances in transform than in fit"
+                    )
+                else:
+                    Xt = [transformers[i].transform(X[i]) for i in range(len(X))]
+            # now we have a list of transformed instances
+
+            # if the output is Series, Xt is a Panel and we convert back
+            if output_scitype == "Series":
+                Xt = convert_to(Xt, to_type=X_input_mtype, as_scitype="Panel")
+
+            # if the output is Primitives, we have a list of one-row dataframes
+            # we concatenate those and overwrite the index with that of X
+            elif output_scitype == "Primitives":
+                Xt = pd.concat(Xt)
+                Xt.index = X.index
```

Comment:
I made a branch from this and when attempting to use the series `SummaryTransformer` the last line of this gave me an error with a panel 3d numpy array input. `TypeError: 'builtin_function_or_method' object is not iterable`

I assume the intention is to retain the indicies from the input, but it appears that the conversion from numpy does not produce anything valid.  Other than that works fine if the line is replaced with i.e. `Xt.index = pd.Index([i for i in range(len(Xt))])`

Reply:
the output should be `pd.DataFrame`, not `numpy` - is that what is failing perhaps?

If not, can you provide reproducible code?

Reply:
The output is a `pd.Dataframe`, it just doesn't like `X.index` for whatever reason.

```
from sktime.datasets import load_unit_test
from sktime.datatypes._panel._convert import from_nested_to_3d_numpy
from sktime.transformations.series.summarize import SummaryTransformer

if __name__ == "__main__":
    X_train, y_train = load_unit_test(split="train", return_X_y=True)
    X_train = from_nested_to_3d_numpy(X_train)
    st = SummaryTransformer()
    print(st.fit_transform(X_train, y_train))
```
Both the 3d numpy array and original dataframe break with the same error for me.

Removing `Xt.index = X.index` or replacing it with `Xt.index = pd.Index([i for i in range(len(Xt))])` fixes this for both cases.

Reply:
fixed with latest comits